File: js/StaticMap.js

fan.afGundam.StaticMap = fan.sys.Obj.$extend(fan.sys.Obj);
fan.afGundam.StaticMap.prototype.$ctor = function() {}
fan.afGundam.StaticMap.prototype.$typeof = function() { return fan.afGundam.StaticMap.$type; }

fan.afGundam.StaticMap.$staticMap = null;
fan.afGundam.StaticMap.staticMap = function() {
	if (fan.afGundam.StaticMap.$staticMap == null) {
		var k = fan.sys.Obj.$type;
		var v = fan.sys.Obj.$type.toNullable();
		fan.afGundam.StaticMap.$staticMap = fan.sys.Map.make(k, v);
	}
	return fan.afGundam.StaticMap.$staticMap;
}

fan.afGundam.StaticMap.get = function(key) {
	return fan.afGundam.StaticMap.staticMap().get(key);
}

fan.afGundam.StaticMap.set = function(key, value) {
	fan.afGundam.StaticMap.staticMap().set(key, value);
}

File: java/StaticMap.java
package fan.afGundam;

import fan.sys.*;
import java.util.Map;
import java.util.HashMap;

public class StaticMap extends FanObj {

	// ---- Construction ------------------------------------------------------

	// constructor factory called by Foo.make
	public static StaticMap make() {
		StaticMap self = new StaticMap();
		make$(self);
		return self;
	}

	// constructor implementation called by subclasses
	public static void make$(StaticMap self) {
		// nuffin'
	}

	// ---- Obj ---------------------------------------------------------------

	private static Type type;

	public Type typeof() {
		if (type == null)
			type = Type.find("afGundam::StaticMap");
		return type;
	}
	
	// ---- Static Data -------------------------------------------------------

	private static Map<Object, Object> staticMap = new HashMap<Object, Object>();
	
	public static Object get(Object key) {
		return staticMap.containsKey(key) ? staticMap.get(key) : null;
	}
	
	public static void set(Object key, Object value) {
		staticMap.put(key, value);
	}
}
